### PR TITLE
Add basic cookies and claims transform samples

### PR DESCRIPTION
**Pull request by HaoK**
Initial sln + port of PR from entropy

@Tratcher

### CODE REVIEW

**Comment by HaoK** [comment]
What kind of versioning layout should we do? 

- samples
-- 1.x (potentially?)
-- 2.0
--- Cookies
--- ClaimsTransform
--- SocialAuth
--- JwtBearer
--- Identity
...
-- 2.1

**Comment by Tratcher** [comment]
I'd prefer to use git history to manage versioning like we do for other repos. Perhaps check in a series of 2.0 samples, drop a label, and then update them to latest dev?

Also a layering question: @natemcmaster @JunTaoLuo If we add this repo to the normal universe build would it be possible to add it after the .All metapackage is built so we could consume it?

File: samples/Cookies/Startup.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc();
+
+            services.AddAuthentication("Cookies") // Sets the default scheme to cookies
```
Comment (Tratcher) [inline samples/Cookies/Startup.cs]: Constants?

**Comment by HaoK** [comment]
Doesn't that make things harder to discover if they have to find the appropriate branch in the repo for the version they want and then look at things?  I guess I would consider this repo more of a docs repo than a library repo, so the focus should be on discover-ability as opposed to focusing soley on latest bits.

**Comment by Tratcher** [comment]
We can set the default branch to the latest release to improve visibility. That way our dev changes don't distract them.

I expect we'll have many projects in this repo even without having duplicates across multiple versions.

**Comment by natemcmaster** [comment]
> If we add this repo to the normal universe build would it be possible to add it after the .All metapackage is built so we could consume it?

At the moment, no, we don't have a way to layer like this. The .All metapackage isn't generated until the very end of the build. We do this because the package isn't useful until you have a runtime store to go with it. If you want to use the .All metapackage in your samples, I suggest leaving this separate from Universe for now.

**Comment by Tratcher** [comment]
.All would be convenient, but it's not required. Making sure these samples aren't broken by API or behavior changes is more important.
As long as we can use the Microsoft.AspNetCore package then the sample code will be the same.

**Comment by JunTaoLuo** [comment]
To build on @natemcmaster's comment, we can't use the .All package now and I don't anticipate we can do this in the future either since we'll still build the .All package after the universe build in the platform world.

**Comment by HaoK** [comment]
Updated moving away from .All reference

**Comment by Tratcher** [comment]
We should add a readme to each project to summarize what we're trying to demo and tell where to find the relevant pieces. E.g. Look at Startup for cookie auth configuration and the AccountController for usage.

We'll also need to think about build and test automation.

**Comment by HaoK** [comment]
These should probably end up working similar to Mvc functional tests (and hopefully can replace music store somewhat)

**Comment by HaoK** [comment]
Looks like this repo is missing all the usual build/test infrastructure/scripts, is there an easy way to add all of that, or should I just copy that from another existing repo?

**Comment by HaoK** [comment]
Bsaically I think we just want what Entropy has...samples and a single test/AuthSamples.FunctionalTest

**Comment by Tratcher** [comment]
Yes, copy the scripts from another repo.

**Comment by HaoK** [comment]
Okay with @natemcmaster 's help getting the build working, I copied over MVC's functional tests which appear to work (in so much as we get a 200 back), we can add real tests later once the samples are in place.  This gives us at least it can compile + run and serve something coverage to start.

File: samples/ClaimsTransformation/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc();
+
+            services.AddAuthentication("Cookies") // Sets the default scheme to cookies
```
Comment (Tratcher) [inline samples/ClaimsTransformation/Startup.cs]: Constants

File: samples/Cookies/ConfigureMyCookie.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        // You can inject services here
+        public ConfigureMyCookie()
+        {
+        }
+
+        public void Configure(string name, CookieAuthenticationOptions options)
+        {
+            // Only configure the schemes you want
+            if (name == "YourScheme")
```
Comment (Tratcher) [inline samples/Cookies/ConfigureMyCookie.cs]: Use the same scheme in Startup

File: test/AuthSamples.FunctionalTests/ClaimsTransformTests.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        public ClaimsTransformTests(SampleTestFixture<ClaimsTransformer.Startup> fixture)
+        {
+            Client = fixture.Client;
+        }
+
+        public HttpClient Client { get; }
+
+        [Fact]
+        public async Task DefaultReturns200()
```
Comment (Tratcher) [inline test/AuthSamples.FunctionalTests/ClaimsTransformTests.cs]: How about one for the MyClaims action?

**Review by Tratcher — COMMENTED** [review]
the samples need their own readme's with short descriptions of the scenario and what files to look in for the relevant pieces.

**Comment by HaoK** [comment]
Updated with readmes and cleanup, I'll file an issue to track adding more functional tests

**Review by Tratcher — APPROVED** [review]

**Comment by HaoK** [comment]
@Tratcher @natemcmaster do I need to anything other than copying over .travis + appveyer to get PR build validation enabled for this repo?

**Comment by Tratcher** [comment]
Only @eilon can do that

**Comment by HaoK** [comment]
Ok I won't worry about that then, merged 6e75c8eb76fea1adfd9b1cd0946a8d1543292ad9

**Pull request closed without merging**